erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -1601,3 +1601,34 @@ def validate_sample_quantity(item_code, sample_quantity, qty, batch_no = None):
 			format(max_retain_qty, batch_no, item_code), alert=True)
 		sample_quantity = qty_diff
 	return sample_quantity
+
+def raw_material_update_on_bom():
+	"""
+	Calculates Average Manufactured Qty(avg_manufactured_qty) for BOM on the Basis of Stock Entries created and
+	it fetches the Stock Entries created within past seven days against BOM.
+	"""
+	past_seven_days = get_date_str(add_days(today(), -7))
```

Comment:
@MyuddinKhatri  Is there any reason it hardcoded to 7 Days.

Reply:
It was the client's requirement that it needs data of the past seven days.

Reply:
@MyuddinKhatri its best practice to it should user-configurable like adding it in stock settings and let the user set days and then compare here instead of hardcoding.

Reply:
@MyuddinKhatri Please make this change